src/channels.rs

```diff
@@ -24,12 +25,21 @@ impl MiEI {
         Ok(())
     }
 
-    pub fn get_role_id(&self, role_name: &str) -> Option<RoleId> {
-        self.courses
-            .values()
-            .filter_map(|x| x.courses.get(role_name))
+    pub fn get_role_id(&self, role_name: &str) -> Vec<RoleId> {
+        let years = &self.courses;
+        let year = Regex::new("ano")
+            .unwrap()
+            .split(role_name)
```

Comment:
The standard library can already handle this: https://doc.rust-lang.org/std/string/struct.String.html#method.split
no need to add the regex crate.

You could instead use the regex to determine the type of role request:
```
if /* regex for year */ {
    years
} else {
    course
}
```